### ISSUE TITLE
Masked Autoencoders Are Scalable Vision Learners

### ISSUE THREAD

**Issue by reyllama** [reyllama/paper-reviews#130]
## TL; DR

- Masked Language Modeling (MLM) has brought enormous success in NLP community (BERT, GPT), but not in computer vision society.
- The differences lies in: (1) Convolutional Architecture (2) Information density between two modalities (3) Role of Decoder
- Mask more proportions of image and design asymmetric auto-encoder

## Motivation

- MLM has been extremely successful in NLP domain. Bring the insights to computer vision domain.

## Previous Work

- BERT
- GPT
- Denoising Autoencoder
- SSL

## Method

![image](https://user-images.githubusercontent.com/46308407/145027333-9c3d88b4-0b97-4a98-ab66-5137685371bc.png)

- Mask roughly 75% of the image patches (non-overlapping patches as in ViT)
- Encoder (ViT) only takes in *unmasked* patches (no <MASK> tokens as well)
- Decoder is much lighter than the encoder, and with encoded tokens and <MASK> tokens, tries to reconstruct the masked regions.
- MSE loss is applied only to the masked regions.

![image](https://user-images.githubusercontent.com/46308407/145027530-7d80d2b5-afdc-48f4-9950-270fcfb0dc8d.png)
![image](https://user-images.githubusercontent.com/46308407/145027559-205cc243-8912-4555-b984-13fbab8b4993.png)

## Experiments

![image](https://user-images.githubusercontent.com/46308407/145027599-80acf928-5763-485a-bf51-d573e7086508.png)

- Masking large portion leads to better performance: image is natural signal with much spatial redundancy.

![image](https://user-images.githubusercontent.com/46308407/145027664-9effc037-fe02-4680-bcaf-4d2a265ce853.png)

- For finetuning, the decoder can be really shallow and thin, but for linear probs, stronger decoder is preferred.
- This is because with smaller decoder, the encoder has to be more specific whereas with strong decoder, encoder can learn more abstract concepts useful for tasks like linear separation.
- Also note that it's more beneficial for the encoder to not take in <MASK> tokens both for performance and computation expense sake.

![image](https://user-images.githubusercontent.com/46308407/145027903-28360307-e769-4045-b2eb-22e32c60d139.png)

- Since lightweight decoder suffice for satisfactory performance, the training can be significantly boosted.

![image](https://user-images.githubusercontent.com/46308407/145028028-04b88cc3-4ce0-494e-ab9f-ec505b622278.png)

- Beats current SOTA on ImageNet-1K evaluation protocol.

![image](https://user-images.githubusercontent.com/46308407/145028075-1053308f-a5da-44a1-81c0-e8eab525da62.png)

- MoCo-v3 performs better for linear probs (more linearly separable representation learning) but with partial finetuning, MAE performs significantly better (*stronger non-linear features*)
- For partial finetuning, they freeze lower layers of the transformer and only update limited layers.

### Transfer on other downstream tasks

![image](https://user-images.githubusercontent.com/46308407/145028243-718e2f5f-de41-405f-811c-4fb3895ca509.png)
![image](https://user-images.githubusercontent.com/46308407/145028328-bc9c136c-599e-40e5-aa72-dacb66857e48.png)

## Comments

- Really good paper. First illustrating the domain difference between vision and NLP (especially in terms of information density) and systematically justify their core design choices (high mask ratio, asymmetric auto-encoder)
- Interesting experiments (partial finetuning - strong but nonlinear features) and useful insights.
- Interesting how such a simple self-sup framework can perform so well. Very readable and insightful.
